File: AnalysePresenceMoustiqueTigre.py
import pandas as pd
import matplotlib.pyplot as plt

# Load your wide-format data
df = pd.read_csv("DataMoustiqueTigre/infectionMoustiqueTigre_data_year.csv")


# Stacked barplot for dengue only (repeat for others if needed)
years = df['year'].astype(str)
auto = df['nb_dept_infected']

# Plotting
plt.figure(figsize=(10, 6))
plt.plot(years, auto, label='nb_dept_infected',marker='o')
plt.title("Nb de départements infectés au cours du temps")
plt.xlabel("Year")
plt.ylabel("Nb de départements ")
plt.xticks(rotation=90)
plt.legend()
plt.tight_layout()
plt.show()





##########################################################" 
# Plots  Max  latitude  of  infected departement  
years = df['year'].astype(str)
auto = df['lat_max_infected']

# Plotting
plt.figure(figsize=(10, 6))
plt.plot(years, auto, label='nlat_max_infected',marker='o')
plt.title("Latitude max de l'infection ")
plt.xlabel("Year")
plt.ylabel("Latitude ")
plt.xticks(rotation=90)
plt.legend()
plt.tight_layout()
plt.show()






##########################################################" 
# Plots  Max  latitude  of  infected departement  
years = df['year'].astype(str)
auto = df['pop_infected']

# Plotting
plt.figure(figsize=(10, 6))
plt.plot(years, auto, label='pop_infected',marker='o')
plt.title("Exposed Population per Year ")
plt.xlabel("Year")
plt.ylabel("Population ")
plt.xticks(rotation=90)
plt.legend()
plt.tight_layout()
plt.show()

File: AnalysePresenceMoustiqueTigre.R
rm(list=ls())
data_Presence = read.csv(file='DataMoustiqueTigre/presenceMoustiqueTigre_data_year.csv')
data_infected_per_year <- read.csv(file='DataMoustiqueTigre/infectionMoustiqueTigre_data_year.csv')



# sum of 1s = number infected
ggplot(data_infected_per_year, aes(x = year, y = nb_dept_infected)) +
  geom_line(color = "red", size = 1.2) +
  geom_point(color = "darkred", size = 2) +
  labs(title = "Number of Infected Departments per Year",
       x = "Year",
       y = "Number of Infected Departments") +
  theme_minimal()

##########################################################" 
# Plots  Max  latitude  of  infected departement  
ggplot(data_infected_per_year, aes(x = year, y = lat_max_infected)) +
  geom_line(color = "red", size = 1.2) +
  geom_point(color = "darkred", size = 2) +
  labs(title = "Lat max infected",
       x = "Year",
       y = "Lat max infected Departments") +
  theme_minimal()


##########################################################" 
# Plots population infected per year   
ggplot(data_infected_per_year, aes(x = year, y = pop_infected)) +
  geom_line(color = "red", size = 1.2) +
  geom_point(color = "darkred", size = 2) +
  labs(title = "Exposed Population per Year",
       x = "Year",
       y = "Exposed Population") +
  theme_minimal()







